### PR TITLE
add middle name to checkout address html templates #8878

### PR DESCRIPTION
**Pull request by AntonEvers**
### Description

Three checkout address templates were lacking the middle name. they are now added.

### Fixed Issues (if relevant)

1. magento/magetno2#8878: middle name is missing from the checkout address html templates

### Manual testing scenarios

1. Create an account with a middle name and log in
2. Add a product to the cart and go to the checkout
3. Visit the shipping address step
4. You should now see the middle name appear in the address name

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by vherasymenko** [comment]
@ajpevers Hi. I was checked your PR and notice next:
1. If user register account with (firstname, middlename, lastname) without filling billing address and goes  "Processed to checkout" field "MiddleName" not filled automatically.
2. If registered user with filled (all fields in his account) on checkout page clicks "New Address"  firstname and lastname fields fills but middlename is empty.

Could you please fix this issues ?? Thanks.

**Comment by AntonEvers** [comment]
@K7Triton working on it

**Comment by AntonEvers** [comment]
Found it

File: app/code/Magento/Checkout/Block/Checkout/AttributeMerger.php [hunk 1]
```diff
@@ -326,6 +326,21 @@ protected function getDefaultValue($attributeCode)
...
                 }
                 break;
+            case 'middlename':
+                if ($this->getCustomer()) {
+                    return $this->getCustomer()->getMiddlename();
+                }
+                break;
+            case 'prefix':
+                if ($this->getCustomer()) {
+                    return $this->getCustomer()->getPrefix();
```
Comment (ishakhsuvarov) [inline app/code/Magento/Checkout/Block/Checkout/AttributeMerger.php]: This PR is on middle name, I would suggest to skip other fields for the sake of testing simplicity.

Comment (AntonEvers) [reply app/code/Magento/Checkout/Block/Checkout/AttributeMerger.php]: I'll create a new PR for prefix and suffix and remove them from this PR.

**Comment by AntonEvers** [comment]
fixes #8878

**Comment by AntonEvers** [comment]
@ishakhsuvarov I've moved the prefix/suffix prefill to https://github.com/magento/magento2/issues/9924

**Review by ishakhsuvarov — APPROVED** [review]

**Comment by ishakhsuvarov** [comment]
@ajpevers Thanks. We will proceed with the review shortly.

**Comment by AntonEvers** [comment]
👍

**Pull request merged**